Implement work-around/fix for incorrect render of manually added HOTP tokens

hicallmeal opened the pull request:
The parsed `<option>` returns the OTP.type as a string, instead of the required integer type. Later in execution, the code can't handle the string in comparisons, etc. 

As a result, selecting a different type causes the resulting entry to be handled as a standard TOTP type. Giving it the appearance and behaviour as such - it also affects backups (..`&counter=undefined`) if done so in the same session. 
Any type of operation apart from the default type (as that's assigned an integer by default) with the option is affected. 
![image](https://github.com/Authenticator-Extension/Authenticator/assets/76943372/83ce20ef-3eb7-43ef-bfc7-87ec9d044491)


In saying that, once reloaded, some other part of the code corrects the entry and storage. 


This just parseInts the string so that it's correctly handled.
You might have some better way of doing this, perhaps earlier on, etc.

hicallmeal committed:
Implement work-around/fix for incorrect render of manually added HOTP tokens

The parsed `<option>` returns the OTP.type in string value, instead of the required integer type. Later in execution, the code can't handle the string in comparisons, etc. 

This just parseInts the string so that it's correctly handled.

mymindstorm reviewed (approved):
Good catch!

mymindstorm commented:
Could you fix the CI issue and manually test the changes? `npm run chrome` -> `chrome://extensions` should allow you to do so.

hicallmeal committed:
updated patch with proper fix

Given the issue derives from the `option`s returning the value as a string by default, the 'number' modifier typecasts the value as a number. 

This then correctly flows on to the script execution
https://v2.vuejs.org/v2/guide/forms.html

hicallmeal commented:
I've updated the patch according to commit notes; essentially in order to return an integer, a `.number` modifier is required. 
https://v2.vuejs.org/v2/guide/forms.html#number

Have tested and compiled successfully.
